### ISSUE TITLE
Templates: Defined Encoding

### ISSUE THREAD

**Issue by FriedrichWeinmann** [PowershellFrameworkCollective/PSModuleDevelopment#67]
- Explicitly specify output encoding
 - Include encoding information in file
 - Override template provided encoding

**Comment by Splaxi** [comment]
Would that fix my issue with `Invoke-PSMDTemplate CommandTest` that doesn't give me the same encoding like `Set-PSMDEncoding`?

I run `Set-PSMDEncoding` against all my ps1 files, this includes my pester test files. When I afterwards want to update the pester file, the encoding is changed.

**Comment by Splaxi** [comment]
And the default encoding should be UTF 8 BOM?

**Comment by FriedrichWeinmann** [comment]
Indeed. UTF8 for widest compatibility and the closest we have for a standard.
BOM so we _know_ rather than _assume_ the encoding of a file
